### ISSUE TITLE
rowwise() + mutate gives crazy results with any()

### ISSUE THREAD

**Issue by antoine-lizee** [hadley/dplyr#1580]
```
data.frame(A = rep(FALSE, 3),
           B = c(NA, rep(FALSE,2)),
           C = rep(FALSE, 3))  %>% 
  rowwise() %>% mutate(D = any(A, B, C))

#Source: local data frame [3 x 4]
#Groups: <by row>
#
#      A     B     C     D
#  (lgl) (lgl) (lgl) (lgl)
#1 FALSE    NA FALSE    NA
#2 FALSE FALSE FALSE FALSE
#3 FALSE FALSE FALSE    NA   <- !
```

**Comment by antoine-lizee** [comment]
Fixed as the latest dev version with commit 4f2d7f8959f888002d46c91592d5800cf9d8e786

**Comment by antoine-lizee** [comment]
@hadley another good reason to release soon :-).
